src/search/algorithms/equivalence_relation.h

```diff
@@ -27,120 +30,61 @@ class Block {
     friend class EquivalenceRelation;
     BlockListIter it_intersection_block;
 public:
-    bool empty() const;
-    ElementListIter insert(int element);
-    void erase(ElementListIter it);
-    ElementListIter begin() {return elements.begin();}
-    ElementListIter end() {return elements.end();}
-    ElementListConstIter begin() const {return elements.begin();}
-    ElementListConstIter end() const {return elements.end();}
+    bool empty() const {
```

Comment:
I think we inlined these methods when we used a templated block class. Now that the class is no longer templated, I'd move the implementations back to the cc file unless you see a significant hit to performance.

Reply:
They were inlined in the old implementation (and never in the implementation file), we just didn't touch this. Since refining equivalence relations indeed can be a bottle neck, as we found out, I would actually leave it like this (or make it one-liners again if you prefer).